### PR TITLE
Refresh type3 audio

### PR DESCRIPTION
**Pull request by oetherington**
- Add server shell commands for deleting/refreshing type3 audio for posts
- Automatically refresh post audio when a user is deleted or changes their name

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207814160565417) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/callbacks/userCallbacks.tsx [hunk 1]
```diff
@@ -501,3 +502,11 @@ getCollectionHooks("Users").editSync.add(function syncProfileUpdatedAt(modifier,
   }
   return modifier;
 });
+
+getCollectionHooks("Users").editAsync.add(async function updatingPostAudio(newUser: DbUser, oldUser: DbUser) {
+  const deletedChanged = newUser.deleted !== oldUser.deleted;
+  const nameChanged = newUser.displayName !== oldUser.displayName;
+  if (nameChanged || deletedChanged) {
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/userCallbacks.tsx:509]: I would check for the presence of API keys here. No need to warn later on if they aren't configured.

**Review by jpaddison3 — APPROVED** [review]

**Comment by jpaddison3** [comment]
Does LW want the behavior on user modification?

**Comment by oetherington** [comment]
@b0b3rt This functionality requires a new API key from type3 (you can see the setting in the new `server/type3.ts` file). If you want this feature you can get a key from them, otherwise just leave the key blank and the update will be disabled.

File: packages/lesswrong/server/type3.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { Globals } from "@/lib/vulcan-lib";
+import { Posts } from "@/lib/collections/posts";
+import { captureEvent } from "@/lib/analyticsEvents";
+
+/* Currently unused
+const type3ClientIdSetting = new DatabaseServerSetting<string | null>('type3.clientId', null)
+const type3WebhookSecretSetting = new DatabaseServerSetting<string | null>('type3.webhookSecret', null)
+*/
+const type3ApiTokenSetting = new DatabaseServerSetting<string | null>("type3.apiToken", null);
+const type3SourceUrlSetting = new DatabaseServerSetting<string>("type3.sourceUrl", "");
```
Comment (oetherington) [inline packages/lesswrong/server/type3.ts:12]: For future notice:

- `type3.apiToken` is a new API token from type3
- `type3.sourceUrl` is set to `https://forum.effectivealtruism.org` (this is needed in order to build absolute prod URLs from a dev machine running the script)

**Pull request merged**